src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -441,6 +446,8 @@ def _check_generic_reject_reasons(self):
 
     def _check_lvm_reject_reasons(self):
         rejected = []
+        if not self.is_acceptable_device_as_lvm_mode:
+            rejected.append("Device type is not acceptable as LVM mode OSD. It should be raw device or partition")
```

Comment:
If I'm not mistaken, this if statement should live nested under the else branch further down https://github.com/ceph/ceph/pull/34966/files#diff-6993397b6f0532424603186ca62822bcR455

If the device has a VG we don't really care about the device type.